mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -72,13 +72,24 @@ class ExecutorStateManagerImpl implements ExecutorStateManager {
         })
         .build();
 
-    @RequiredArgsConstructor
+    @Getter
     @ToString
-    static class JobRequirements {
-        public final Map<Double, Integer> coresToWorkerCount;
+    class JobRequirements {
+        private final Map<TaskExecutorGroupKey, Integer> groupToTaskExecutorCount;
+
+        JobRequirements(Map<SchedulingConstraints, Integer> constraintsToTaskExecutorCount) {
+            this.groupToTaskExecutorCount = constraintsToTaskExecutorCount
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(entry -> findBestFitGroupOrDefault(entry.getKey()), Map.Entry::getValue));
+        }
 
         public int getTotalWorkers() {
-            return coresToWorkerCount.values().stream().mapToInt(Integer::intValue).sum();
+            return groupToTaskExecutorCount.values().stream().mapToInt(Integer::intValue).sum();
+        }
+
+        private TaskExecutorGroupKey findBestFitGroupOrDefault(SchedulingConstraints constraints) {
+            return findBestGroup(constraints).orElse(new TaskExecutorGroupKey(constraints.getMachineDefinition(), constraints.getSchedulingAttributes()));
```

Comment:
Does falling back to default here means there is no fitting TE ever registered? Is that a legit case?

Reply:
shall we log something then?